docs/resources/credential.md

```diff
@@ -0,0 +1,92 @@
+---
+subcategory: "Unity Catalog"
+---
+# databricks_credential Resource
+
+-> This resource can only be used with a workspace-level provider.
+
+-> This feature is in [Public Preview](https://docs.databricks.com/release-notes/release-types.html).
+
+A credential represents an authentication and authorization mechanism for accessing services on your cloud tenant. Each credential is subject to Unity Catalog access-control policies that control which users and groups can access the credential.
+
+To create credentials, you must be a Databricks account admin or have the `CREATE SERVICE CREDENTIAL` privilege. The user who creates the credential can delegate ownership to another user or group to manage permissions on it
```

Comment:
Will `create service credential` work with service principals on azure?

Reply:
Should we also mention `CREATE_STORAGE_CREDENTIAL` ?

Reply:
our sdk go struct does not match the OpenAPI spec for some reason...

Reply:
@farrucosanjurjo-db wdyt?

Reply:
> our go-sdk struct does not match the OpenAPI spec for some reason...maybe we need to wait for it to be regenerated

I think that's the issue here. We recently updated the proto files (and hence the OpenAPI specs) to include the management of storage credentials with the `/credentials` endpoints

Reply:
> Will create service credential work with service principals on azure?

No. We only support Azure Managed Identities

Reply:
> Should we also mention CREATE_STORAGE_CREDENTIAL ?

yes, we should. See the docs https://docs.databricks.com/api/workspace/credentials/createcredential